hivemq-edge/src/main/java/com/hivemq/configuration/reader/BridgeExtractor.java

```diff
@@ -37,52 +36,92 @@
 import com.hivemq.configuration.entity.bridge.RemoteSubscriptionEntity;
 import com.hivemq.configuration.entity.listener.tls.KeystoreEntity;
 import com.hivemq.configuration.entity.listener.tls.TruststoreEntity;
-import com.hivemq.configuration.service.BridgeConfigurationService;
 import com.hivemq.edge.HiveMQEdgeConstants;
 import com.hivemq.exceptions.UnrecoverableException;
+import com.hivemq.util.Topics;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
-import com.hivemq.util.Topics;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import javax.inject.Inject;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
+import java.util.function.Consumer;
 import java.util.stream.Collectors;
 
-public class BridgeConfigurator implements Syncable<List<MqttBridgeEntity>>{
+public class BridgeExtractor implements ReloadableExtractor<List<@NotNull MqttBridgeEntity>, List<@NotNull MqttBridge>> {
 
-    private static final Logger log = LoggerFactory.getLogger(BridgeConfigurator.class);
+    private static final Logger log = LoggerFactory.getLogger(BridgeExtractor.class);
     public static final String KEYSTORE_TYPE_PKCS12 = "PKCS12";
     public static final String KEYSTORE_TYPE_JKS = "JKS";
 
-    private final @NotNull BridgeConfigurationService bridgeConfigurationService;
+    private volatile @NotNull List<@NotNull MqttBridge> bridgeEntities = List.of();
+    private volatile @Nullable Consumer<List<@NotNull MqttBridge>> bridgeEntitiesConsumer = cfg -> log.debug("No consumer registered yet");
 
-    private volatile List<MqttBridgeEntity> configEntity;
-    private volatile boolean initialized = false;
+    private final @NotNull ConfigFileReaderWriter configFileReaderWriter;
 
-    @Inject
-    public BridgeConfigurator(
-            final @NotNull BridgeConfigurationService bridgeConfigurationService) {
-        this.bridgeConfigurationService = bridgeConfigurationService;
+    public BridgeExtractor(@NotNull final ConfigFileReaderWriter configFileReaderWriter) {
+        this.configFileReaderWriter = configFileReaderWriter;
     }
 
+    public synchronized void addBridge(final @NotNull MqttBridge mqttBridge) {
+        if (!mqttBridge.isPersist()) {
+            log.info(
+                    "MQTT Bridge '{}' has persist flag set to false, QoS for publishes from local subscriptions will be downgraded to AT_MOST_ONCE.",
+                    mqttBridge.getId());
+        }
+
+        bridgeEntities = new ImmutableList.Builder<MqttBridge>()
+                .addAll(bridgeEntities)
+                .add(mqttBridge)
+                .build();
+
+        notifyConsumer();
+        configFileReaderWriter.writeConfigWithSync();
+    }
+
+    public @NotNull List<MqttBridge> getBridges() {
+        return new ImmutableList.Builder<MqttBridge>()
+                .addAll(bridgeEntities)
+                .build();
```

Comment:
If `bridgeEntities` is already immutable, why do we create another immutable list?